### ISSUE TITLE
Make sure the package is compatible with split subpackages later. 

### ISSUE THREAD

**Issue by johann-petrak** [GateNLP/python-gatenlp#12]
We may want to create namespace subpackages later, so try to be compatible with that 
from the start.

See https://packaging.python.org/guides/packaging-namespace-packages/

E.g. separate packages that implement nlp tools for languages, come with pre-trained models etc.

**Comment by johann-petrak** [comment]
As far as I understand this, the recommended, python 3 approach to doing this is to create a namespace package that must not have an `__init__.py` file. In other words, it is not possible to have a normal package and also use the same package name as a namespace package for additional subpackages. 

This would mean that all we have now would need to move to something like gatenlp.core instead of just gatenlp so we can later add subpackages like gatenlp.ml gatenlp.visualize or similar.

**Comment by johann-petrak** [comment]
The alternative would be to just use package names like `gatenlp_ml` or `gatenlp_visualize` in addition to `gatenlp`.

**Comment by johann-petrak** [comment]
Check out how package options work, e.g. "dask[complete]"

**Comment by johann-petrak** [comment]
This is called package "extras" and can be used to install optional dependencies. 

So if there is a module already in gatenlp that requires package x, we could setup things so that x is only installed if we do `pip install gatenlp[x]`. 

Extras can be specified in `setup.py`:
* setup.py: use `extras_require` e.g. `extras_require={"stanza": ["stanza>=x.y", "someotherpackage"]}` 

We should move from multiple requirements.txt to using  the `extras_require ` feature!

**Comment by johann-petrak** [comment]
Regarding subpackages: there is no proper way at all to add a new subpackage to an existing package which already has an implementation - namespace packages need to be special and need a structure where only the subpackages have an implementation. 

So lets just give up on this and accept that all new packages for extending gatenlp would have to live in a different root package, e.g. `gatenlp_ml` or `gatenlp_german` or the like. On the plus side this means that installation of "gatenlp_ml"  would also add package "gatenlp_ml"  and not "gatenlp.ml" . 

So basically, we either have to add stuff to the main "gatenlp" package, or implement things in a different root package. In some cases it may help to make the gatenlp package aware of separate root packages and use them, e.g. over extras. 

Closing this now.

**Comment by johann-petrak** [comment]
UPDATE: apparently it SHOULD be possible to install into an existing package tree later, IF the new distribution does not override any existing init file (and probably also not anything else). 
So if the new distribution only provides a subpackage and only the init file for that subpacakge (and depends on the main package) it actually should work as expected!
